dipy/denoise/gibbs.py

```diff
@@ -0,0 +1,314 @@
+           Analysis and their Application to the Healthy Ageing Brain
+           (Doctoral thesis). https://doi.org/10.17863/CAM.29356
+    .. [2] Kellner E, Dhital B, Kiselev VG, Reisert M. Gibbs-ringing artifact
+           removal based on local subvoxel-shifts. Magn Reson Med. 2016
+           doi: 10.1002/mrm.26054.
+    """
+    if np.any(G0) is None or np.any(G1) is None:
+        G0, G1 = _weights(image.shape)
+
+    img_c1 = _gibbs_removal_1d(image, axis=1, n_points=n_points)
+    img_c0 = _gibbs_removal_1d(image, axis=0, n_points=n_points)
+
+    C1 = np.fft.fft2(img_c1)
+    C0 = np.fft.fft2(img_c0)
+    imagec = abs(np.fft.ifft2(np.fft.fftshift(C1)*G1 + np.fft.fftshift(C0)*G0))
+
+    return imagec
+
+
+def gibbs_removal(vol, slice_axis=2, n_points=3):
+    """ Suppresses gibbs ringing artefacts of images volumes.
+
+    Parameters
+    ----------
+    vol : array ([X, Y, Z]) or ([X, Y, Z, g])
+        Matrix containing one volume (3D) or multiple (4D) volumes of images.
+    slice_axis : int (0, 1, or 2)
+        Data axis corresponding to the number of acquired slices. Default is
+        set to the third axis
+    n_points : int, optional
+        Number of neighbour points to access local TV (see note). Default is
+        set to 3.
+
+    Returns
+    -------
+    vol : array ([X, Y, Z]) or ([X, Y, Z, g])
+        Matrix containing one volume (3D) or multiple (4D) volumes of corrected
+        images.
+
+    Notes
+    -----
+    For 4D matrix last element should always correspond to the number of
+    diffusion gradient directions.
+
+    References
+    ----------
+    Please cite the following articles
+    .. [1] Neto Henriques, R., 2018. Advanced Methods for Diffusion MRI Data
+           Analysis and their Application to the Healthy Ageing Brain
+           (Doctoral thesis). https://doi.org/10.17863/CAM.29356
+    .. [2] Kellner E, Dhital B, Kiselev VG, Reisert M. Gibbs-ringing artifact
+           removal based on local subvoxel-shifts. Magn Reson Med. 2016
+           doi: 10.1002/mrm.26054.
+    """
+
+    # Make sure that slice are ordering in matrix axis 2
+    if slice_axis > 2:
+        raise ValueError("Different slices have to be organized along" +
+                         "one of the 3 first matrix dimensions")
+    elif slice_axis < 2:
+        vol = np.swapaxes(vol, slice_axis, 2)
+
+    # check matrix dimension
+    nd = vol.ndim
+    if nd == 4:
+        inishap = vol.shape
+        vol = vol.reshape((inishap[0], inishap[1], inishap[2]*inishap[3]))
+    elif nd > 4:
+        raise ValueError("Data have to be a 4D, 3D or 2D matrix")
+    elif nd < 2:
+        raise ValueError("Data is not an image")
+
+    # Produce weigthing functions for 2D gibbs removal
+    shap = vol.shape
+    G0, G1 = _weights(shap[:2])
+
+    # Run gibbs removal of 2D images
+    if nd == 2:
+        vol = _gibbs_removal_2d(vol, n_points=n_points, G0=G0, G1=G1)
```

Comment:
True! However, to do this change I will need to add two extra if statements, one in "check matrix dimension" and another in "Reshape data to original format". Since my code is not crashing for 2D image I will leave the code as it is.